yoti_test.go

```diff
@@ -52,10 +52,40 @@ func TestYotiClient_KeyLoad_Failure(t *testing.T) {
 		},
 	}
 
-	_, _, errorStrings := client.getActivityDetails(encryptedToken)
+	_, _, err := client.getActivityDetails(encryptedToken)
 
-	assert.Assert(t, len(errorStrings) > 0)
-	assert.Check(t, strings.HasPrefix(errorStrings[0], "Invalid Key"))
+	assert.Check(t, err != nil)
+	assert.Check(t, strings.HasPrefix(err.Error(), "Invalid Key"))
+	type Interface interface {
+		Temporary() bool
+	}
+	_, temporary := err.(Interface)
```

Comment:
You could represent this like so

```go
_, temporary := err.(interface{Temporary() bool})
```

Which just seems a bit neater to me.

Also we should be checking the value returned by Temporary on the temporary error since it is possible for Temporary to return false in the case of Temporary errors returned by the go standard libs.